op-node/rollup/derive/span_channel_out.go

```diff
@@ -152,9 +161,11 @@ func (co *SpanChannelOut) AddSingularBatch(batch *SingularBatch, seqNum uint64)
 
 	// if the compressed data *plus* the new rlp data is under the target size, return early
 	// this optimizes out cases where the compressor will obviously come in under the target size
-	rlpGrowth := co.activeRLP().Len() - co.lastCompressedRLPSize
-	if uint64(co.compressed.Len()+rlpGrowth) < co.target {
-		return nil
+	if !SkipOptimize {
+		rlpGrowth := co.activeRLP().Len() - co.lastCompressedRLPSize
+		if uint64(co.compressed.Len()+rlpGrowth) < co.target {
+			return nil
+		}
 	}
```

Comment:
I am not in favor of flags which change component behavior for unit tests. There should be other ways of testing by using a different `rng`, thus making differently sized structures which won't have the issue you described?

Reply:
Good point, let me try.